ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/BallerinaShell.java

```diff
@@ -118,30 +116,27 @@ public void run() {
                 String source = terminal.readLine(leftPrompt, rightPrompt).trim();
                 start = Instant.now();
                 if (!commandHandler.handle(source)) {
-                    ShellCompilation shellCompilation = evaluator.getCompilation(source);
-                    Optional<PackageCompilation> compilation = shellCompilation.getPackageCompilation();
-                    if (ExceptionStatus.SUCCESS == shellCompilation.getExceptionStatus()) {
-                        Optional<ShellReturnValue> shellReturnValue = evaluator.getValue(compilation);
-                        if (shellReturnValue.isPresent() &&
-                                shellReturnValue.get().getExceptionStatus() == ExceptionStatus.SUCCESS) {
-                            result = shellReturnValue.get().getResult();
-                            terminal.result(result);
-                        } else if (shellReturnValue.isPresent() && shellReturnValue.get().getExceptionStatus() ==
+                    Optional<ShellReturnValue> shellReturnValue = evaluator.execute(source);
+                    if (shellReturnValue.isPresent() && ExceptionStatus.SUCCESS ==
+                            shellReturnValue.get().getExceptionStatus()) {
+                        result = shellReturnValue.get().getResult();
+                        terminal.result(result);
+                    } else if (shellReturnValue.isPresent() && shellReturnValue.get().getExceptionStatus() ==
                                 ExceptionStatus.INVOKER_FAILED) {
                             if (isContainsUndefinedModules(evaluator.diagnostics())) {
                                 currentStateDiagnostics = List.copyOf(evaluator.diagnostics());
                                 evaluator.resetDiagnostics();
                                 executeChanges(source, currentStateDiagnostics);
                             }
-                        }
-                    } else if (shellCompilation.getExceptionStatus() == ExceptionStatus.INVOKER_FAILED) {
+                    } else if (shellReturnValue.get().getExceptionStatus() == ExceptionStatus.INVOKER_FAILED) {
                         if (isContainsUndefinedModules(evaluator.diagnostics())) {
                             currentStateDiagnostics = List.copyOf(evaluator.diagnostics());
                             evaluator.resetDiagnostics();
                             executeChanges(source, currentStateDiagnostics);
                         }
                     }
```

Comment:
Can't we improve this if-else by getting `shellReturnValue.get()` into a variable and using early return if `shellReturnValue.isPresent` is false?

Reply:
Added in ea054445b72a422651c331b20cc9c2bb6056a8f8